rustler/src/types/binary.rs

```diff
@@ -338,10 +357,24 @@ impl<'a> Binary<'a> {
     /// If `offset + length` is out of bounds, an error will be returned.
     pub fn make_subbinary(&self, offset: usize, length: usize) -> NifResult<Binary<'a>> {
         let min_len = length.checked_add(offset);
-        if min_len.ok_or(Error::BadArg)? > self.inner.size {
+        if min_len.ok_or(Error::BadArg)? > self.size {
             return Err(Error::BadArg);
         }
 
+        Ok(unsafe { self.make_subbinary_unchecked(offset, length) })
+    }
+
+    /// Returns a new view into the same binary.
+    ///
+    /// This method is an unsafe variant of `Binary::make_subbinary` in that it does not check for
+    /// `offset + length < self.len()` and always returns a `Binary`.
+    ///
+    /// # Safety
+    ///
+    /// If `offset + length` is out of bounds, this call results in *undefined behavior*. The
+    /// caller has to ensure that `offset + length < self.len()`.
+    #[allow(unused_unsafe)]
+    pub unsafe fn make_subbinary_unchecked(&self, offset: usize, length: usize) -> Binary<'a> {
```

Comment:
Btw, should it be called `_unchecked` or `_unsafe` for consistency? I am just wondering what is the idiom here. :)

Reply:
My understanding is that everything that just skips a bounds check is called `unchecked`, everything that has some arbitrary unsafe-ness is called `unsafe` :)